### ISSUE TITLE
(SURE-6076) AKS TF-provisioned private cluster import is failing

### ISSUE THREAD

**Issue by kkaempf** [rancher/aks-operator#169]
### Issue description:
AKS private cluster import fails with the error: "cannot detect cluster [xxxx] upstream DNS prefix"

### Business impact:
Not being able to manage AKS clusters through Rancher.

### Troubleshooting steps:

Import a private AKS cluster using the Rancher new AKS cluster Registration command=-> KO (dns error msg)
Create a new private AKS cluster using Rancher UI, and used 'az command-invoke' to register the AKS cluster in Rancher> KO (same error msg)
Create a new private AKS cluster using Azure portal, used Rancher UI to import the cluster without success. Used 'az command-invoke' from az cli, to import the cluster, with no success.
Create a new private AKS cluster using TF, used the Rancher UI to import the cluster => KO 
Verified that Rancher is reachable (curl to Rancher_Server URL and name resolvable), from within a pod of a newly TF-provisioned private AKS cluster
Set Rancher in Debug mode (showed the error message, but no more evidences)

### Repro steps:

Provision a private AKS cluster TF, using  azure_rm_provider, using a custom vnet and custom private DNS zone
Import the cluster into Rancher from the UI

### Workaround:
Is a workaround available and implemented? yes
What is the workaround: Import the AKS clusters as custom clusters

### Actual behavior:
AKS private clusters cannot be imported into Rancher.

### Expected behavior:
AKS private clusters with custom networking (existing vnet) and custom private DNS zone can be imported into Rancher.

### Files, logs, traces:
Attached is the Rancher debug log, traces of DNS and AKS reachability from on-prem, and the Rancher reachability test from Azure.

### Additional notes:

Customer's Rancher server is hosted in an on-premises environment, with an on-prem network.
AKS clusters are private, and deployed in private existing networks (vnet) and custom DNS zone.  On-premises network is connected to Azure networks via ExpressRoute
AKS clusters can be directly managed from the customer's on-prem environment using kubectl with the kubeconfig credentials downloaded via az cli.

**Comment by mjura** [comment]
I have got requested configuration files, I am investigating it once again

**Comment by mjura** [comment]
Fix was submitted https://github.com/rancher/aks-operator/pull/178

**Comment by mjura** [comment]
Command which I used to build AKS cluster:

```
az aks create -g mjura-eastus -n mjura-outbound -l eastus \
      --node-count 2 --network-plugin kubenet --network-policy calico \
      --vnet-subnet-id "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXX/..." \
      --enable-managed-identity --assign-identity "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXX/..." \
      --load-balancer-sku standard --outbound-type userDefinedRouting \ 
      --ssh-key-value "/home/azureuser/.ssh/id_rsa.pub" \
      --enable-private-cluster --private-dns-zone "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXX/..." \
      --disable-public-fqdn --fqdn-subdomain "test-subdomain-k8sapi"

```

**Comment by cpinjani** [comment]
**Ticket https://github.com/rancher/aks-operator/issues/169  - Test Results - :white_check_mark:**

**Setup**
Create AKS Private cluster using custom VNet & custom DNS Private Zone and import the same into Rancher UI

Issue reproduced on unfixed version:
![image](https://github.com/rancher/aks-operator/assets/73099870/769608d1-c432-472c-98ca-a451c7e66881)

Issue fixed, validated on: `v2.7-head - 5b9667d`
After cluster import, upgrade K8s version, add/delete node pools worked fine.

![image](https://github.com/rancher/aks-operator/assets/73099870/c6ed2d97-fb67-43ea-87e6-c128b23cf5c2)
